<!-- This file is originally pr2_2dnav/pr2_2dnav.launch --> 
<launch>
  <arg name="ROBOT" default="$(optenv ROBOT pr2)"/>
  <arg name="gazebo" default="false" />
  <arg name="launch_look_forward" default="true"/>
  <arg name="launch_teleop_joystick" default="true" />
  <arg name="launch_navigation" default="true" />

  <include file="$(find pr2_machine)/$(arg ROBOT).machine" />

  <group if="$(arg launch_teleop_joystick)">
    <include file="$(find jsk_pr2_startup)/jsk_pr2_move_base/pr2_teleop_general_joystick.launch">
      <arg name="ROBOT" value="$(arg ROBOT)"/>
    </include>
    <include file="$(find jsk_pr2_startup)/jsk_pr2_move_base/safe_teleop.xml" />
  </group>

  <group if="$(arg launch_navigation)">
    <include file="$(find jsk_pr2_startup)/jsk_pr2_move_base/amcl_node.xml" />
    <include file="$(find jsk_pr2_startup)/jsk_pr2_sensors/lasers_and_filters.xml">
      <arg name="gazebo" value="$(arg gazebo)"/>
    </include>
    <include file="$(find jsk_pr2_startup)/jsk_pr2_move_base/ground_plane.xml" />
    <include file="$(find jsk_pr2_startup)/jsk_pr2_move_base/move_base.xml" />

    <!-- To ignore tilt-laser -->
    <node name="tilt_laser_mux" pkg="topic_tools" type="mux"
          args="ground_object_cloud_mux ground_object_cloud empty_cloud">
      <remap from="mux" to="tilt_laser_mux"/>
    </node>
    <node name="empty_cloud_publisher" pkg="jsk_pr2_startup"
          type="publish_empty_cloud.py">
      <param name="frame_id" value="/laser_tilt_link" />
    </node>

    <!-- look path forward when navigation -->
    <node name="look_forward_in_navigation" pkg="roseus" type="roseus"
          args="$(find jsk_pr2_startup)/jsk_pr2_move_base/look-forward-in-nav.l"
          machine="c2" respawn="true"
          if="$(arg launch_look_forward)"/>
  </group>
</launch>
